### PR TITLE
Make pylint survive WebUI artifacts

### PR DESCRIPTION
**Pull request by VladimirSlavik**
tl;dr: SSIA.

As part of the current WebUI workflow, we need some other repositories, they contain python files, and our pylint script is greedy and checks them too. To make problems worse, the repos also contain symlinks that point to nowhere, until you use the scripts from these repos. Then some symlinks start pointing to ISOs. And of course, we try to detect python scripts by reading the whole file and finding hashbangs. This PR tries to fix this bag of problems as folows:

* Skip files that can't be opened,
* ~~Skip files that are over 10 MB,~~ Read only first line up to 1024 characters when detecting hashbangs,
* Add false positive matches for the repo subdirs.

### CODE REVIEW

File: tests/pylint/runpylint.py [hunk 1]
```diff
@@ -39,6 +39,10 @@ def __init__(self):
             FalsePositive(r"E1101.*: Instance of 'int' has no 'name_from_node' member"),
             FalsePositive(r"E1101.*: Instance of 'int' has no 'generate_backup_passphrase' member"),
             FalsePositive(r"E1101.*: Instance of 'int' has no 'dasd_is_ldl' member"),
+
+            # Cockpit stuff that gets checked out inside our repo
+            FalsePositive(r"ui/webui/bots"),
```
[the remark this answers is not in the document]
Comment (KKoukiou) [reply tests/pylint/runpylint.py:44]: And once you add that directory you can remove https://github.com/rhinstaller/anaconda/blob/master/tests/pylint/pylintrc#L11

File: tests/pylint/runpylint.py [hunk 2]
```diff
@@ -68,18 +68,26 @@ def _get_py_paths(self, directory):
...
 
             for f in files:
+                file_path = root + "/" + f
                 try:
-                    with open(root + "/" + f) as fo:
+                    if os.path.getsize(file_path) > 10e6:
+                        # Skip too large files - 10 MB should be enough for everyone!
+                        continue
+                    with open(file_path) as fo:
                         lines = fo.readlines()
```
Comment (poncovka) [inline tests/pylint/runpylint.py:77]: Weird. Why do we read all lines since we later work with only the first one? I think we should fix that instead of the checking the size of the file.

File: tests/pylint/runpylint.py [hunk 3]
```diff
@@ -68,18 +68,26 @@ def _get_py_paths(self, directory):
...
                 except UnicodeDecodeError:
                     # If we couldn't open this file, just skip it.  It wasn't
                     # going to be valid python anyway.
                     continue
+                except FileNotFoundError:
+                    # This can happen with ui/webui/bots/images/* that is a symlink to nowhere.
+                    # These files are not Python scripts, so let's not even try.
```
Comment (KKoukiou) [inline tests/pylint/runpylint.py:84]: Is this not respected by the ignore-paths?

Comment (VladimirSlavik) [reply tests/pylint/runpylint.py:84]: No, we first scan here in this function, then pass that list to pylint, which filters by its own ignores. Yes, that's super ugly. Why? I don't know, it's already there.

File: tests/pylint/runpylint.py (hunk 2 above)
Comment (VladimirSlavik) [reply tests/pylint/runpylint.py:77]: I agree, and have some ideas, so I'll take a look. Let's stay mindful of scope creep, though. I want to ignore some new stuff, not be forced to rewrite the whole thing because you don't like it.

File: tests/pylint/runpylint.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (VladimirSlavik) [reply tests/pylint/runpylint.py]: Thanks! Done.

File: tests/pylint/runpylint.py (hunk 3 above)
Comment (KKoukiou) [reply tests/pylint/runpylint.py:84]: Doh, thanks!

File: tests/pylint/runpylint.py (hunk 2 above)
Comment (poncovka) [reply tests/pylint/runpylint.py:77]: Definitely. However, I suggested it as an alternative to the `os.path.getsize(file_path) > 10e6` check. It seems better to fix the original issue then to add a workaround for it.

Comment (poncovka) [reply tests/pylint/runpylint.py:77]: This seems to work.

```
     def _get_py_paths(self, directory):
@@ -70,7 +69,7 @@ class AnacondaLintConfig(CensorshipConfig):
             for f in files:
                 try:
                     with open(root + "/" + f) as fo:
-                        lines = fo.readlines()
+                        line = fo.readline()
                 except UnicodeDecodeError:
                     # If we couldn't open this file, just skip it.  It wasn't
                     # going to be valid python anyway.
@@ -78,7 +77,7 @@ class AnacondaLintConfig(CensorshipConfig):
 
                 # Test any file that either ends in .py or contains #!/usr/bin/python
                 # in the first line.
-                if f.endswith(".py") or (lines and str(lines[0]).startswith("#!/usr/bin/python")):
+                if f.endswith(".py") or (line and str(line).startswith("#!/usr/bin/python")):
                     retval.append(root + "/" + f)
 
         return retval
```

Comment (VladimirSlavik) [reply tests/pylint/runpylint.py:77]: Thank you! I missed that method. That will work, as it has also a data limit parameter. I was looking for a way to limit the amount of data read, in case you have an ISO file and read it until first line end...

**Review by poncovka — APPROVED** [review]
Thanks!

**Comment by jkonecny12** [comment]
I really like this new solution thanks!

File: tests/pylint/pylintrc [hunk 4]
```diff
@@ -8,7 +8,13 @@ extension-pkg-whitelist=
 # Add files or directories to the denylist. They should be base names, not
 # paths.
 # This is auto-fetched from cockpit shared library
```
Comment (VladimirSlavik) [inline tests/pylint/pylintrc:10]: Oh. Forgot to remove this comment, or move it below.
